src/EphemeralConnector.js

```diff
@@ -138,34 +141,48 @@ class EphemeralConnector extends BaseConnector {
    * claim before it.
    */
   async get (link, did = null, privkey = null) {
-    let reference = BaseConnector.referenceFromLink(link)
-    let pubkey = BaseConnector.referenceFromDid(did)
-
-    let signature
-    if (pubkey != null && privkey != null) {
-      let identity = await this.identityFactory.fromDid(did, privkey)
-      signature = identity.sign(reference)
-    }
+    try {
+      let retrievedObjFromCache = this.myCache.get(link, true)
```

Comment:
If you use the option without passing errorOnMissing, you can instead check for undefined, return it if it's not and compute the value otherwise. 

This prevents the entire function from being nested in the catch, which, in my opinion makes it more readable.